Driver/Driver/Scenes/LoginScreen/ViewController/LoginViewController.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  LoginViewController.swift
+//  Driver
+//
+//  Created by Fatih Doğan on 22.02.2023.
+//
+
+import UIKit.UIViewController
+
+final class LoginViewController: BaseViewController {
+    
+    // MARK: Outlets
+    @IBOutlet private weak var userNameTextField: UITextField!
+    @IBOutlet private weak var passwordTextField: UITextField!
+    
+    // MARK: Properties
+    var viewModel: LoginViewModelProtocol!
+    
+    // MARK: LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        userNameTextField.delegate = self
+        passwordTextField.delegate = self
+    }
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        userNameTextField.becomeFirstResponder()
+    }
+    // MARK: Functions
+    @IBAction private func loginButtonPressed(_ sender: UIButton) {
+        viewModel.login(userName: userNameTextField.text, password: passwordTextField.text)
+    }
+}
+extension LoginViewController: LoginViewDelegate {
+    func handleOutput(_ output: LoginViewModelOutput) {
+        switch output {
+        case .error(let errorDescription):
+            showAlert(alertType: .caution, for: errorDescription, actionName: "Tamam")
+        case .setSpinner(let isLoading):
+            isLoading ? Animator.shared.showAnimation(viewController: self) : Animator.shared.hideAnimation()
+        case .loginSuccesfully:
+            viewModel.navigateHomeView()
+        }
+    }
+}
+extension LoginViewController: UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        if textField == userNameTextField {
+            userNameTextField.endEditing(true)
+            passwordTextField.becomeFirstResponder()
+        } else {
+            textField.endEditing(true)
+        }
+        return true
+    }
+}
+extension LoginViewController {
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        view.endEditing(true)
```

Comment:
We should call super as the first thing here, [as suggested for the method](https://developer.apple.com/documentation/uikit/uiresponder/1621142-touchesbegan)